mypy/checkstrformat.py

```diff
@@ -136,38 +144,155 @@ def __init__(self, type: str,
         # Full formatted expression (i.e. key plus following attributes and/or indexes).
         # Used only for str.format() calls.
         self.field = field
-        self.whole_seq = whole_seq
 
     @classmethod
-    def from_match(cls, match: Match[str],
-                   non_standard_spec: bool = False) -> 'ConversionSpecifier':
+    def from_match_format_call(cls, match: Match[str], start_pos: int = -1,
```

Comment:
Style nit: I don't expect this to be ever subclassed, so `@staticmethod` would seem more reasonable (actually I'd personally use a module-level function). Using `@classmethod` and `cls(...)` kind of gives the message that this can be subclassed.

Reply:
I'm using a simpler approach now